libafl/src/observers/map.rs

```diff
@@ -747,13 +747,12 @@ where
     type Entry = T;
 
     #[inline]
-    fn get(&self, pos: usize) -> &T {
-        &self.as_slice()[pos]
+    fn get(&self, pos: usize) -> T {
+        self.as_slice()[pos]
     }
 
-    #[inline]
-    fn get_mut(&mut self, idx: usize) -> &mut T {
-        &mut self.as_mut_slice()[idx]
+    fn set(&mut self, pos: usize, val: T) {
+        self.map.as_mut_slice()[pos] = val;
     }
```

Comment:
This is generally a good change. In your exact application, this will likely be quite poor perf (since it introduces a "lock"/"unlock" pair every access).

Reply:
As an aside, an `#[inline]` disappeared here. The compiler should inline it anyways when appropriate, so maybe these should all be removed. For now, please re-add it for consistency with the rest of the type.